test/jdk/tools/launcher/Arrrghs.java

```diff
@@ -601,6 +601,38 @@ void testMainMethod() throws FileNotFoundException {
         tr.checkPositive();
         if (!tr.testStatus)
             System.out.println(tr);
+
+        //private method with parameter, usable method without parameter:
+        createJar(new File("some.jar"), new File("Foo"),
+            "private static void main(String[] args){}",
+            "void main() {System.out.println(\"THE_CHOSEN_ONE\");}");
+        tr = doExec(javaCmd, "-jar", "some.jar");
+        tr.contains("THE_CHOSEN_ONE");
+        if (!tr.testStatus)
+            System.out.println(tr);
+
+        //method with a wrong return type with parameter, usable method without parameter:
+        createJar(new File("some.jar"), new File("Foo"),
+            "public static int main(String[] args){ return -1; }",
+            "void main() {System.out.println(\"THE_CHOSEN_ONE\");}");
+        tr = doExec(javaCmd, "-jar", "some.jar");
+        tr.contains("THE_CHOSEN_ONE");
+        if (!tr.testStatus)
+            System.out.println(tr);
+
+        // instance method abstract class:
+        createJarForSource(null, new File("some.jar"), new File("Foo"),
+                """
+                public abstract class Foo {
+                    void main() {
+                        System.out.println("Cannot be called.");
+                    }
+                }
+                """);
+        tr = doExec(javaCmd, "-jar", "some.jar");
+        tr.contains("Error: abstract class Foo can not instantiated");
+        if (!tr.testStatus)
+            System.out.println(tr);
```

Comment:
Hello Jan, I think this is pre-existing, but it looks like we don't fail the test even when `tr.testStatus` indicates a failure? We seem to only write out to `System.out` and return from the test method. Am I misreading the test code?

Reply:
Thank you for that. I didn't even realize that this was a "main()" test. I saw that there was a `@Test` annotation on these test methods and assumed it was either a testng or junit test. Turns out it's just another internal annotation of the `TestHelper`.

> It is probably not how I would write the test from scratch, but re-writing an already existing test also does not seem like a good option.

What you have in the current PR is fine with me.